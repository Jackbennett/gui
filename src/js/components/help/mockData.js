export const helpProps = {
  docsVersion: '',
  findLocalIpAddress: jest.fn,
  isHosted: false,
  isEnterprise: false,
  menderDebPackageVersion: 'master',
  menderVersion: 'master',
  menderArtifactVersion: 'master',
  org: {
    tenant_token: ''
  }
};
