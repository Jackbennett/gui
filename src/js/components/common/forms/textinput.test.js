import React from 'react';
import { render } from '@testing-library/react';
import TextInput from './textinput';
import { undefineds } from '../../../../../tests/mockData';

describe('TextInput Component', () => {
  it('renders correctly', async () => {
    const { baseElement } = render(<TextInput attachToForm={jest.fn} detachFromForm={jest.fn} />);
    const view = baseElement.firstChild.firstChild;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });
});
