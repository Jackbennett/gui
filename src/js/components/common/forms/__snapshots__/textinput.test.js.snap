// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Component renders correctly 1`] = `
<div
  class="MuiFormControl-root"
>
  <label
    class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
    data-shrink="false"
  />
  <div
    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    style="width: 400px; max-width: 100%;"
  >
    <input
      aria-invalid="false"
      class="MuiInputBase-input MuiInput-input"
      type="text"
      value=""
    />
  </div>
  <p
    class="MuiFormHelperText-root"
    id="component-error-text"
  />
</div>
`;
